plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -655,3 +661,32 @@ func (p *Plugin) admitResourceSlice(nodeName string, a admission.Attributes) err
 
 	return nil
 }
+
+func (p *Plugin) admitCSR(nodeName string, a admission.Attributes) error {
+	// Create requests for Kubelet serving signer and Kube API server client
+	// kubelet signer with a CN that begins with "system:node:" must have a CN
+	// that is exactly the node's name.
+	// Other CSR attributes get checked in CSR validation by the signer.
+	if a.GetOperation() == admission.Create {
+		csr, ok := a.GetObject().(*certapi.CertificateSigningRequest)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+		}
+		if csr.Spec.SignerName != certapi.KubeletServingSignerName && csr.Spec.SignerName != certapi.KubeAPIServerClientKubeletSignerName {
+			return nil
+		}
+		csrBytes := make([]byte, base64.StdEncoding.DecodedLen(len(csr.Spec.Request)))
+		_, err := base64.StdEncoding.Decode(csrBytes, csr.Spec.Request)
+		if err != nil {
+			return admission.NewForbidden(a, errors.New("unable to base64 decode csr"))
+		}
+		x509cr, err := certapi.ParseCSR(csrBytes)
+		if err != nil {
+			return admission.NewForbidden(a, fmt.Errorf("unable to parse csr: %w", err))
+		}
+		if strings.HasPrefix(x509cr.Subject.CommonName, "system:node:") && x509cr.Subject.CommonName != fmt.Sprintf("system:node:%s", nodeName) {
```

Comment:
these two signer names already require "system:node:" prefixes, just require that unilaterally

```suggestion
		if x509cr.Subject.CommonName != fmt.Sprintf("system:node:%s", nodeName) {
```